base_layer/wallet/src/output_manager_service/service.rs

```diff
@@ -411,9 +427,48 @@ where
                 let output_statuses_by_tx_id = self.get_output_info_by_tx_id(tx_id)?;
                 Ok(OutputManagerResponse::OutputInfoByTxId(output_statuses_by_tx_id))
             },
+            OutputManagerRequest::CreateOwnerProof(output, nonce) => self
+                .create_owner_proof(output, nonce)
+                .await
+                .map(OutputManagerResponse::OwnerProofCreated),
         }
     }
 
+    async fn create_owner_proof(
+        &self,
+        commitment: Commitment,
+        message: String,
+    ) -> Result<RistrettoComSig, OutputManagerError> {
+        let output = self.resources.db.fetch_by_commitment(commitment.clone())?;
+        let wallet_output: WalletOutput = output.into();
+        let private_key = self
+            .resources
+            .key_manager
+            .get_private_key(&wallet_output.spending_key_id)
+            .await?;
+        let nonce_a = RistrettoSecretKey::random(&mut OsRng);
+        let nonce_x = RistrettoSecretKey::random(&mut OsRng);
+
+        let nonce_commitment = self.resources.factories.commitment.commit(&nonce_x, &nonce_a);
+        let challenge = Blake2b::<U64>::new()
+            .chain_update(commitment.as_bytes())
+            .chain_update(nonce_commitment.as_bytes())
+            .chain_update(format!("create_owner_proof::{}", message).as_bytes())
+            .finalize();
```

Comment:
I realize this PR is still a draft, but I'd recommend using the existing domain-separated hash functionality instead of this message formatting approach if possible.

Reply:
Yeah I got this working with what was available. This code is actually copied from the examples in taricrypto. I'm not really happy with it, but this is the API that ComSig exposes. Ideally the ComSig signing api should be updated to match the schnorr api, where the domain is part of the call

Reply:
Right, and I know there's a longstanding [issue](https://github.com/tari-project/tari-crypto/issues/159) you filed to improve the API that was sidelined due to not using that signature until now. Even so, you should be able to generate the challenge hash using a domain-separated hasher anyway, no? Then you can ensure it'll take care of things like domain separation and input collisions automagically.